packages/headless/src/controllers/facets/range-facet/numeric-facet/headless-numeric-facet.ts

```diff
@@ -59,7 +141,13 @@ export function buildNumericFacet(
     'buildNumericFacet'
   );
 
-  dispatch(registerNumericFacet(options));
+  if (!options.generateAutomaticRanges && options.currentValues === undefined) {
+    engine.logger.error(
+      'currentValues should be specified for buildNumericFacet when generateAutomaticRanges is false.'
+    );
+  }
```

Comment:
The previous `NumericFacetOptions` used a discriminating union to force `currentValues` to be specified when `generateAutomaticRanges` was false. Unfortunately, we can't use a discriminating union anymore.

Reply:
I think this is a good compromise. I still think a compile-time alert is more dev friendly, but we can revert to that once the parser can resolve multiple types. 👍